File: java/src/main/java/com/svix/agentic_commerce_protocol/TotalType.java
// This file is @generated
package com.svix.agentic_commerce_protocol;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.agentic_commerce_protocol.Utils.ToQueryParam;

public enum TotalType implements ToQueryParam {
    ITEMS_BASE_AMOUNT("items_base_amount"),
    ITEMS_DISCOUNT("items_discount"),
    SUBTOTAL("subtotal"),
    DISCOUNT("discount"),
    FULFILLMENT("fulfillment"),
    TAX("tax"),
    FEE("fee"),
    TOTAL("total");
    private final String value;

    TotalType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/TotalType.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class TotalType : ToQueryParam {
    @SerialName("items_base_amount") ITEMS_BASE_AMOUNT,
    @SerialName("items_discount") ITEMS_DISCOUNT,
    @SerialName("subtotal") SUBTOTAL,
    @SerialName("discount") DISCOUNT,
    @SerialName("fulfillment") FULFILLMENT,
    @SerialName("tax") TAX,
    @SerialName("fee") FEE,
    @SerialName("total") TOTAL;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
